### PR TITLE
Custom Route shapes fix

### LINKED ISSUE #2595: Unable to provide custom route line shapes
Methods like `NavigationMapViewDelegate.navigationMapView(_:shapeFor:)` and `NavigationMapViewDelegate.navigationMapView(_:simplifiedShapeFor:)` were originally added to allow the developer to essentially provide a custom route line colored by something other than traffic congestion. However, as of #2377, these methods are no longer called by NavigationMapView, so the route line can only be colored by traffic congestion. When these methods are implemented, we should use them instead of creating a route line colored by a gradient.

/cc @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by Udumft**
Fixes #2595 

Restored calling `NavigationMapViewDelegate.navigationMapView(_:simplifiedShapeFor:)` method for shape customization and verified that custom shapes mechanism works.

There is a discussion going on around these methods on how to improve the API and make customization easy and straightforward. Eventually we should remove these methods, but decided to leave them for now because there is no alternative ideas developed atm and some clients already use that API.

### CODE REVIEW

File: MapboxNavigation/NavigationMapView.swift [hunk 1]
```diff
@@ -473,7 +473,8 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         let allRoutesShape = navigationMapViewDelegate?.navigationMapView(self, shapeFor: routes) ?? shape(for: routes, legIndex: legIndex)
         let allRoutesSource = addAllRoutesSource(style, shape: allRoutesShape)
         
-        let mainRouteCasingShape = navigationMapViewDelegate?.navigationMapView(self, shapeFor: [mainRoute]) ?? shape(for: [mainRoute], legIndex: legIndex)
+        let mainRouteCasingShape = navigationMapViewDelegate?.navigationMapView(self, simplifiedShapeFor: mainRoute) ?? shape(forCasingOf: mainRoute, legIndex: legIndex)
```
Comment (1ec5) [inline MapboxNavigation/NavigationMapView.swift:476]: Should `allRoutesShape` also call `navigationMapView(_:simplifiedShapeFor:)`? What should call `navigationMapView(_:shapeFor:)`? Previously, a developer would’ve implemented `navigationMapView(_:shapeFor:)` to provide the shape data for the whole route (normally to be used as the casing) and `navigationMapView(_:simplifiedShapeFor:)` to provide the shape data for the route broken up by arbitrary segments (normally to be colored by traffic congestion).

Comment (Udumft) [reply MapboxNavigation/NavigationMapView.swift:476]: As I found in the code, `navigationMapView(_:shapeFor:)` is meant to provide route shapes, and `navigationMapView(_:simplifiedShapeFor:)` meant to provide the casing shape. At least when I provided my custom Shapes like that they were colored as route and casing respectfully. Even if my custom shape for `navigationMapView(_:shapeFor:)` had just one `Polyline` it still was colored with congestion segments. 
Doesn't it supposed to be like that?

Comment (1ec5) [reply MapboxNavigation/NavigationMapView.swift:476]: > As I found in the code, `navigationMapView(_:shapeFor:)` is meant to provide route shapes, and `navigationMapView(_:simplifiedShapeFor:)` meant to provide the casing shape.

Yes, sorry, I got the two methods mixed up in https://github.com/mapbox/mapbox-navigation-ios/pull/2623#discussion_r488007899. If you provide only one shape in `navigationMapView(_:shapeFor:)`, I don’t think there should be colored congestion segments.

**Review by IodaMikeMapbox — APPROVED** [review]

File: MapboxNavigation/NavigationMapView.swift (hunk 1 above)
Comment (Udumft) [reply MapboxNavigation/NavigationMapView.swift:476]: So... Is this convo resolved? :)

**Review by 1ec5 — APPROVED** [review]
Remember to add a changelog entry about restoring the ability to customize the route shape.

**Comment by MaximAlien** [comment]
@1ec5, @Udumft After merging this PR to `main` I can see that route casing is no longer present:

Screen Shot 2020-10-28 at 3 28 10 PM

Before route line casing was visible:

Screen Shot 2020-10-28 at 3 28 37 PM

When I check `shape(forCasingOf:legIndex:)` implementation I see that `MGLPolylineFeature` doesn't use `isAlternateRoute`. Since `shape(forCasingOf:legIndex:)` is called only main route line in default case shouldn't we set `polyline.attributes["isAlternateRoute"] = false`?

**Comment by Udumft** [comment]
Shame on me for missing this one :( but thank you for pointing that out! Fix submitted in [mapbox/mapbox-navigation-ios/pull/2715](https://github.com/mapbox/mapbox-navigation-ios/pull/2715)

**Review by Rmkasun-8 — COMMENTED** [review]
Navigation on mapview

**Review by Rmkasun-8 — COMMENTED** [review]
Change

**Review by Rmkasun-8 — COMMENTED** [review]
Authored

**Pull request merged**